File: RelatingDataFieldCustom/RelatingDataFieldCustom.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.UI;
using Telerik.Sitefinity.Web.UI.Fields;

namespace SitefinityWebApp.RelatingDataFieldCustom
{
    public class RelatingDataFieldCustom : RelatingDataField
    {
        //public override IEnumerable<System.Web.UI.ScriptReference> GetScriptReferences()
        //{
        //    var baseScripts = base.GetScriptReferences().ToList();
        //    baseScripts.Add(new System.Web.UI.ScriptReference("~/RelatingDataFieldCustom/RelatingDataFieldCustom.js"));

        //    return baseScripts;
        //}

        public override IEnumerable<System.Web.UI.ScriptReference> GetScriptReferences()
        {
            var scripts = new List<ScriptReference>(base.GetScriptReferences());
            scripts.Add(new ScriptReference("Telerik.Sitefinity.Web.UI.Fields.Scripts.ILocalizableFieldControl.js", "Telerik.Sitefinity"));
            scripts.Add(new ScriptReference("Telerik.Sitefinity.Web.UI.Fields.Scripts.IRelatingDataField.js", "Telerik.Sitefinity"));
            scripts.Add(new ScriptReference("Telerik.Sitefinity.Web.UI.Fields.Scripts.RelatingDataField.js", "Telerik.Sitefinity"));

            scripts.Add(new System.Web.UI.ScriptReference("~/RelatingDataFieldCustom/RelatingDataFieldCustom.js"));

            return scripts;
        }

        //protected override string ScriptDescriptorType
        //{
        //    get
        //    {
        //        return typeof(RelatingDataFieldCustom).FullName;
        //    }
        //}

        //protected override string ScriptDescriptorTypeName
        //{
        //    get
        //    {
        //        return typeof(RelatingDataFieldCustom).FullName;
        //    }
        //}
    }
}

File: RelatingDataFieldCustom/RelatingDataFieldCustom.js
/// <reference name="MicrosoftAjax.js"/>
Type.registerNamespace("SitefinityWebApp.RelatingDataFieldCustom");

SitefinityWebApp.RelatingDataFieldCustom.RelatingDataFieldCustom = function (element) {

    SitefinityWebApp.RelatingDataFieldCustom.RelatingDataFieldCustom.initializeBase(this, [element]);
};

SitefinityWebApp.RelatingDataFieldCustom.RelatingDataFieldCustom.prototype = {

    /* --------------------------------- set up and tear down --------------------------------- */

    initialize: function () {
        SitefinityWebApp.RelatingDataFieldCustom.RelatingDataFieldCustom.callBaseMethod(this, 'initialize');

        requirejs.config({
            paths: {
                FlatContentSelector: '/RelatingDataFieldCustom/FlatContentSelectorCustom',
                RelatedDataGridTemplate: '/SFHtml/SitefinityWebApp.RelatingDataFieldCustom.RelatedDataGrid.sfhtml',
                FlatContentSelectorOriginal: 'Telerik.Sitefinity.RelatedData.Web.UI.Scripts.FlatContentSelector'
            }
        });
    },
    dispose: function () {
        SitefinityWebApp.RelatingDataFieldCustom.RelatingDataFieldCustom.callBaseMethod(this, 'dispose');
    }
};

SitefinityWebApp.RelatingDataFieldCustom.RelatingDataFieldCustom
    .registerClass('SitefinityWebApp.RelatingDataFieldCustom.RelatingDataFieldCustom',
    Telerik.Sitefinity.Web.UI.Fields.RelatingDataField);